### PR TITLE
Adds import of new Estimation and Prediction analyses

### PR DESCRIPTION
**Pull request by wivern**
Adds import of new Estimation and Prediction analyses

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/service/client/atlas/AtlasClient2_7.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import feign.Param;
+import feign.RequestLine;
+import java.util.List;
+import org.springframework.http.MediaType;
+
+public interface AtlasClient2_7 extends AtlasClient {
+	@RequestLine("GET /prediction")
+	List<PatientLevelPredictionInfo> getPatientLevelPredictions();
+
+	@RequestLine("GET /prediction/{id}/download?packageName={package}")
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/datanode/service/client/atlas/AtlasClient2_7.java]: The same comment as [here](https://github.com/OHDSI/ArachneNodeAPI/pull/302#discussion_r278578972)

Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/service/client/atlas/AtlasClient2_7.java]: Totally agree, this should be reviewed and merged with new CC and Pathways imports

File: src/main/java/com/odysseusinc/arachne/datanode/config/HandlebarsConfig.java [hunk 2]
```diff
@@ -37,6 +37,8 @@
     private static final String ESTIMATION_RUNNER_TEMPLATE = "estimation/runner.mustache";
     private static final String PATIENT_LEVEL_PREDICTION_RUNNER_TEMPLATE = "plp/main.r";
     private static final String COHORT_CHARACTERIZATION_RUNNER_TEMPLATE = "cc/main.r";
+    private static final String PREDICTION_RUNNER_TEMPLATE = "plp/runAnalysis.R"; //for Atlas 2.7.0+
+    private static final String NEW_ESTIMATION_RUNNER_TEMPLATE = "estimation/runAnalysis.R"; //for Atlas 2.7.0+
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/datanode/config/HandlebarsConfig.java]: I would use consistent naming: both PLE and PLP have old and new versions

File: src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/EstimationListItem.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package com.odysseusinc.arachne.datanode.dto.atlas;
+
+import java.io.Serializable;
+import java.util.Date;
+
+public class EstimationListItem extends BaseAtlasEntity implements Serializable {
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/EstimationListItem.java]: Why not `extends CommonAnalysis`? - `createdBy`, `createdDate`, etc overlap

File: src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/ExpressionDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import java.io.IOException;
+import java.util.Objects;
+
+public class ExpressionDeserializer extends JsonDeserializer<String> {
+	@Override
+	public String deserialize(JsonParser parser, DeserializationContext context) throws IOException, JsonProcessingException {
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/ExpressionDeserializer.java]: Some comment regarding logic would be very appreciated

Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/dto/serialize/ExpressionDeserializer.java]: In older Atlas versions expression was a `String` containing json inside, but starting from version *2.8* it becomes an object. `ExpressionDeserializer` is required to remain consistent w/o changing internal logic of DataNode.

File: src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/EstimationListItem.java (hunk 3 above)
Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/EstimationListItem.java]: It should be both a `BaseAtlasEntity` to build map between entities in multiple Atlas instances and `CommonAnalysis`. But `Java` is not `Scala`, there no `traits` here.

**Comment by pavgra** [comment]
PLE list request fails:
image

**Comment by pavgra** [comment]
The issue is related to `atlasClientPool`. When I started Data Node, it had old info about an Atlas instance and instantiated AtlasClient v2.5 and then the Atlas version was refreshed to v2.7 but pool was still returning AtlasClient v2.5 (cached)

**Comment by pavgra** [comment]
Import should work properly with non-secure Atlas instances:
```
2019-04-30 12:03:54.661 ERROR 45236 --- [NodeScheduler-3] c.o.a.d.service.impl.CohortServiceImpl   : Process List Requests checking failure, Can not construct instance of com.odysseusinc.arachne.datanode.dto.atlas.AtlasUserDTO: no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: java.io.BufferedReader@decee31; line: 1, column: 152] (through reference chain: java.util.ArrayList[0]->com.odysseusinc.arachne.datanode.dto.atlas.EstimationAnalysis["createdBy"]) reading GET http://api.ohdsi.org/WebAPI/estimation
```

**Comment by pavgra** [comment]
`AtlasClient2_5` is instantiated for Atlas with version `{"version":"2.7.2.bcbs"}`. Should be `AtlasClient2_7`

**Comment by pavgra** [comment]
One more issue:
```
2019-04-30 13:26:50.670 ERROR 32216 --- [NodeScheduler-3] c.o.a.d.service.impl.CohortServiceImpl   : Process List Requests checking failure, No converter found capable of converting from type [com.odysseusinc.arachne.datanode.dto.atlas.ComparativeCohortAnalysis] to type [com.odysseusinc.arachne.commons.api.v1.dto.CommonEstimationDTO]
```

**Comment by pavgra** [comment]
Old Circe version is used:
```
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "PlaceOfServiceLocation" (class org.ohdsi.circe.cohortdefinition.VisitOccurrence), not marked as ignorable (13 known properties: "Gender", "VisitSourceConcept", "CorrelatedCriteria", "VisitType", "ProviderSpecialty", "CodesetId", "VisitLength", "VisitTypeExclude", "PlaceOfService", "OccurrenceEndDate", "OccurrenceStartDate", "First", "Age"])
 at [Source: java.io.StringReader@2aa3d236; line: 1, column: 71638] (through reference chain: org.ohdsi.circe.cohortdefinition.CohortExpression["CensoringCriteria"]->java.lang.Object[][5]->org.ohdsi.circe.cohortdefinition.VisitOccurrence["PlaceOfServiceLocation"])
```

**Comment by pavgra** [comment]
`runAnalysis` scripts suffer from the same issues which have been already resolved in similar scripts in Atlas:
- Need to install packages in `no-multiarch` mode
- Issue with results folder for PLE

Additionally, `install_local` doesn't work properly for Windows (the same issue still exists in Atlas).

**Comment by pavgra** [comment]
Cannot import PLE which contains special chars in its name:
```
error save file to disk, filename=f6fd0c74-13e2-409a-aba4-e5c7dbddb02d ex=java.nio.file.InvalidPathException: Illegal char <:> at index 19: Graham replication : Comparative effectiveness of dabigatran vs warfarin for risk of stroke_target.ohdsi.sql
```

**Pull request merged**